### PR TITLE
 refactor!,feat: Replace `NonEmptyVec` with `Vec1`

### LINKED ISSUE #427: refactor/question: Rename `NonEmptyVec` to `Vec1`?
There is a case now that requires two elements. If we do the rename, we can use `Vec1`, `Vec2`, ...

### PR DESCRIPTION
**Pull request by duesee**
Closes #427 

### TODO

* [x] Add a test for `Vec<T, N` for `N != 1`

### CODE REVIEW

File: imap-types/src/core.rs [hunk 1]
```diff
@@ -1475,12 +1469,34 @@ impl<T> IntoIterator for NonEmptyVec<T> {
...
+    }
+}
+
+/// A `Vec` that always contains >= 2 elements.
+///
+/// Some messages in IMAP require a list of at least one element. We encoded these situations in a
+/// non-empty vector type to not produce invalid messages.
+///
+/// The `Debug` implementation equals `Vec` with an attached `{2,}` at the end.
+pub type Vec2<T> = VecN<T, 2>;
```
Comment (jakoschiko) [inline imap-types/src/core.rs:1498]: Hm, I'm not sure whether I like the name. Based on the name I'd assume that it has exactly 2 elements. Like the `Vec2` from the `glam` crate.

Alternatives:
- `Min2Vec`
- `Vec2Plus`
- `VecAtLeast2`
- `Vec2OrMoah`

File: imap-types/src/core.rs [hunk 2]
```diff
@@ -1399,31 +1399,31 @@ impl<'a> AsRef<str> for Charset<'a> {
...
-///
-/// Some messages in IMAP require a list of at least one element. We encoded these situations in a
-/// non-empty vector type to not produce invalid messages.
-///
-/// The `Debug` implementation equals `Vec` with an attached `+` at the end.
 #[cfg_attr(feature = "bounded-static", derive(ToStatic))]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(Clone, PartialEq, Eq, Hash)]
-pub struct NonEmptyVec<T>(pub(crate) Vec<T>);
+pub struct VecN<T, const N: usize>(pub(crate) Vec<T>);
```
Comment (jakoschiko) [inline imap-types/src/core.rs:1405]: It looks like you couldn't decide whether you make this type public or not.

I'd would prefer to document `VecN` extensively and replace the doc of `Vec1` and `Vec2` with a short comment.

File: imap-types/src/core.rs
Comment (jakoschiko) [inline imap-types/src/core.rs:1]: `impl<T; const N: usize> From<[T; N]> for VecN<T, N>` would also be useful.

File: imap-types/src/core.rs [hunk 3]
```diff
@@ -1475,12 +1469,34 @@ impl<T> IntoIterator for NonEmptyVec<T> {
...
+
+impl<T> From<T> for Vec1<T> {
+    fn from(value: T) -> Self {
+        VecN(vec![value])
+    }
+}
+
+/// A `Vec` that always contains >= 2 elements.
+///
+/// Some messages in IMAP require a list of at least one element. We encoded these situations in a
```
[the remark this answers is not in the document]
Comment (duesee) [reply imap-types/src/core.rs:1494]:
```suggestion
/// Some messages in IMAP require a list of at least two elements. We encoded these situations in a
```

File: imap-types/src/core.rs (hunk 1 above)
Comment (duesee) [reply imap-types/src/core.rs:1498]: I see what you mean but still lean towards keeping `Vec2`. nom [uses](https://docs.rs/nom/latest/nom/multi/fn.separated_list1.html), e.g., `separated_list{0,1}` with the meaning >= 0 and >= 1. You may need to look it up, but given that I expect people to be often exposed to `Vec1`, maybe it's okay they need to look it up once? imap-types already has quite some "typing required" (and rightward drift). So -- while usually a bad argument! -- it makes a short name like `Vec2` hard to beat here... What do you think?

File: imap-types/src/core.rs (hunk 2 above)
Comment (duesee) [reply imap-types/src/core.rs:1405]: Hehe, that's exactly what happend :-) Couldn't publicly alias a private type (of course). Good catch!

File: imap-types/src/core.rs (hunk 1 above)
Comment (jakoschiko) [reply imap-types/src/core.rs]: Same for Vec1 and VecN. But no strong feelings from my side.

File: imap-types/src/core.rs (hunk 2 above)
Comment (jakoschiko) [reply imap-types/src/core.rs:1414]: doc(hidden) would be an option. But I like VecN and would prefer to keep it and document it properly.

**Pull request merged**